src/xhr.js

```diff
@@ -55,7 +67,11 @@ class Xhr {
    * @return {!Promise<!JSONValue>}
    */
   fetchJson(input, opt_init) {
-    return this.fetch_.call(null, input, opt_init).then(response => {
+    if (opt_init && opt_init.method &&
+        opt_init.method.toUpperCase() == 'POST') {
+        setupJsonPost_(opt_init);
```

Comment:
Thinking, shouldn't we also add `'Accept': 'application/json'` to GET JSON requests? Maybe instead we can do this:

1. Immediately default init: `init = opt_init || {}`
2. Default init.method
3. Call `setupJson_()` (or just inline this code here - depending on tests)
4. In the `setupJson_` check the method and set headers/body as appropriate